drivers/serial/uart_stm32.c

```diff
@@ -901,7 +901,11 @@ static void uart_stm32_isr(const struct device *dev)
 	} else if (LL_USART_IsEnabledIT_RXNE(UartInstance) &&
 			LL_USART_IsActiveFlag_RXNE(UartInstance)) {
 		/* clear the RXNE by flushing the fifo, because Rx data was not read */
+#ifdef USART_SR_RXNE
+		LL_USART_ClearFlag_RXNE(UartInstance);
+#else
```

Comment:
Comment "clear the RXNE by flushing the fifo, because Rx data was not read" should be moved to the else part

Reply:
done